src/server/contest/runner.py

```diff
@@ -77,44 +74,43 @@ def check_result(self, input_file):
             # compilation/runtime error
             self.tests.append(result['error'])
 
-    def execute(self, file_path, input_file_path):
+    def execute(self, file_path: Path, input_file_path: Path):
         """Run files on local computer.
         file_path is the name of the file with absolute path """
 
-        # file_path cotains the name of the file with path and extension
-        file_path_without_extension = os.path.splitext(file_path)[0]
-        # contains directory part of file_path
-        file_dir = os.path.dirname(file_path)
-        # name of the file
-        file_name = os.path.basename(file_path)
-        # name of the file without extension
-        file_name_without_extension = file_name.split('.')[0]
+        # # file_path cotains the name of the file with path and extension
+        # file_path_without_extension = file_path.parent / file_path.stem
+        # # contains directory part of file_path
+        # file_dir = file_path.parent
+        # # name of the file
+        # file_name = file_path.name
+        # # name of the file without extension
+        # file_name_without_extension = file_path.stem
         # executable is stored in /sandbox/jail/executable
         EXECUTABLE_FILE = "executable_" + str(self.submission.id)
-        executable_path = os.getcwd() + "/contest/sandbox/jail/" + \
-            EXECUTABLE_FILE
-
-        # printing for testing purposes
-        # print ("file_path: "+file_path)
-        # print ("file_path_without_extension: "+file_path_without_extension)
-        # print ("file dir : "+file_dir)
-        # print ("file name : "+file_name)
-        # print ("file name without extension : "+file_name_without_extension)
-        # print("executable file directory:" + executable_path)
+        executable_path = JAIL_DIR / EXECUTABLE_FILE
+        JAIL_DIR.mkdir(parents=True, exist_ok=True)  # if the dir doesn't exist
+        # # printing for testing purposes
```

Comment:
Same here. Since you're already sweeping the code, it'd be fine to remove these.

Reply:
ok then i'll remove the unnecessary lines.